File: fxgl-base/src/main/java/com/almasb/fxgl/gameplay/AchievementProgressEvent.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.gameplay;

import javafx.event.EventType;

/**
 * Fired when a numeric value based achievement has made some progress.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
public final class AchievementProgressEvent extends AchievementEvent {

    public static final EventType<AchievementProgressEvent> PROGRESS =
            new EventType<>(ANY, "PROGRESS");

    private double value, max;

    public double getValue() {
        return value;
    }

    public double getMax() {
        return max;
    }

    public AchievementProgressEvent(Achievement achievement, double value, double max) {
        super(PROGRESS, achievement);
        this.value = value;
        this.max = max;
    }

    @Override
    public String toString() {
        return "AchievementProgressEvent[value=" + value + ",max=" + max + "]";
    }
}

File: fxgl-achievement/src/main/kotlin/com/almasb/fxgl/achievement/AchievementProgressEvent.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.achievement

import javafx.event.EventType

/**
 * Fired when a numeric value based achievement has made some progress.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
class AchievementProgressEvent(
        achievement: Achievement,
        val value: Double,
        val max: Double
) : AchievementEvent(PROGRESS, achievement) {

    companion object {
        val PROGRESS = EventType<AchievementProgressEvent>(AchievementEvent.ANY, "PROGRESS")
    }

    override fun toString() = "AchievementProgressEvent[value=$value,max=$max]"
}